libdnf-plugins/actions/actions.cpp

```diff
@@ -0,0 +1,725 @@
+std::pair<std::string, bool> Actions::substitute(
+    const libdnf::base::TransactionPackage * trans_pkg,
+    const libdnf::rpm::Package * pkg,
+    std::string_view in,
+    std::filesystem::path file,
+    int line_number) {
+    auto & logger = *base.get_logger();
+    std::string ret;
+    bool error = false;
+    size_t pos = 0;
+    do {
+        auto var_pos = in.find("${", pos);
+        ret += in.substr(pos, var_pos - pos);
+        if (var_pos == std::string::npos) {
+            break;
+        }
+        var_pos += 2;
+        auto var_end_pos = in.find('}', var_pos);
+        if (var_end_pos == std::string::npos) {
+            logger.error(
+                "Actions plugin: Syntax error: Incomplete variable name \"{}\" in file \"{}\" at line {}",
+                std::string(in.substr(var_pos - 2)),
+                file.native(),
+                line_number);
+            error = true;
+            break;
+        }
+
+        const auto var_name = in.substr(var_pos, var_end_pos - var_pos);
+        std::optional<std::string> var_value;
+        if (var_name == "pid") {
+            var_value = std::to_string(getpid());
+        } else if (var_name.starts_with("conf.")) {
+            auto config_opts = base.get_config().opt_binds();
+            auto it = config_opts.find(std::string(var_name.substr(5)));
+            if (it != config_opts.end()) {
+                var_value = it->second.get_value_string();
+            }
+        } else if (var_name.starts_with("var.")) {
+            auto vars = base.get_vars();
+            try {
+                var_value = vars->get_value(std::string(var_name.substr(4)));
+            } catch (std::out_of_range &) {
+            }
+        } else if (var_name.starts_with("tmp.")) {
+            if (auto it = tmp_variables.find(std::string(var_name.substr(4))); it != tmp_variables.end()) {
+                var_value = it->second;
+            }
+        } else if (var_name.starts_with("pkg.")) {
+            auto pkg_key = var_name.substr(4);
+            if (pkg) {
+                if (pkg_key == "name") {
+                    var_value = pkg->get_name();
+                } else if (pkg_key == "arch") {
+                    var_value = pkg->get_arch();
+                } else if (pkg_key == "version") {
+                    var_value = pkg->get_version();
+                } else if (pkg_key == "release") {
+                    var_value = pkg->get_release();
+                } else if (pkg_key == "epoch") {
+                    var_value = pkg->get_epoch();
+                } else if (pkg_key == "repo_id") {
+                    var_value = pkg->get_repo_id();
+                } else if (pkg_key == "license") {
+                    var_value = pkg->get_license();
+                } else if (pkg_key == "location") {
+                    var_value = pkg->get_location();
+                } else if (pkg_key == "vendor") {
+                    var_value = pkg->get_vendor();
+                }
```

Comment:
How about adding the following `pkg` attributes: `"na"`, `"evr"`, `"nevra"`. These can be constructed from the current attributes, but it's inconvenient and these can be needed quite often especially the `"nevra"` is tedious to put together.

Reply:
OK